src/governance.cpp

```diff
@@ -778,6 +717,50 @@ CGovernanceObject::CGovernanceObject(const CGovernanceObject& other)
     fExpired = other.fExpired;
 }
 
+void CGovernanceObject::SetMasternodeInfo(const CTxIn& vin, const CPubKey& pubkey)
+{
+    vinMasternode = vin;
+    pubkeyMasternode = pubkey;
+}
+
+bool CGovernanceObject::Sign(CKey& keyMasternode)
+{
+    LOCK(cs);
+    CPubKey pubKeyCollateralAddress;
+    CKey keyCollateralAddress;
+
+    std::string strError;
+    uint256 nHash = GetHash();
+    std::string strMessage = nHash.ToString();
+
+    if(!darkSendSigner.SignMessage(strMessage, vchSig, keyMasternode)) {
+        LogPrintf("CGovernanceObject::Sign -- SignMessage() failed\n");
+        return false;
+    }
+
+    if(!darkSendSigner.VerifyMessage(pubkeyMasternode, vchSig, strMessage, strError)) {
+        LogPrintf("CGovernanceObject::Sign -- VerifyMessage() failed, error: %s\n", strError);
+        return false;
+    }
+
+    return true;
+}
+
+bool CGovernanceObject::IsSignatureValid()
```

Comment:
Pls change name to `CheckSignature`, see https://github.com/dashpay/dash/pull/939#issuecomment-239363080